### DISCUSSION TITLE
Improve recovery from temporary socket.io connection lost

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#1761] [Ideas / Feature Requests]
If the connection somehow gets interrupted shortly, socketio will trigger a reconnect. But the handshake fails and triggers a full page reload.

Reproduction:

```py
@ui.page('/')
async def index():
    ui.button('drop connection', on_click=lambda: ui.run_javascript('socket.io.engine.close()', respond=False))
```

It would be great to recover without reloading the full page.

**Comment by rodja** [comment]
I started an possible implementation in pull request https://github.com/zauberzeug/nicegui/pull/1762.

**Reply by falkoschindler** [reply]
The PR has been released. I'll close this discussion.

**Comment by rodja** [comment]
Coincidence! Right after I drafted the first idea on how to fix it, we received a question describing this exact problem with Google Cloud Run: https://github.com/zauberzeug/nicegui/discussions/1763
